source/ring_buffer.c

```diff
@@ -214,7 +214,16 @@ int aws_ring_buffer_acquire_up_to(
 }
 
 static inline bool s_buf_belongs_to_pool(const struct aws_ring_buffer *ring_buffer, const struct aws_byte_buf *buf) {
-    return buf->buffer >= ring_buffer->allocation && buf->buffer + buf->capacity <= ring_buffer->allocation_end;
+/* comparison of pointers to different objects is undefined (C11 6.5.8) */
+#ifdef CBMC
+#    pragma CPROVER check push
+#    pragma CPROVER check disable "pointer"
```

Comment:
Is there any way to ensure that all CBMC proofs that call this function call it with pointers to the same object, instead of disabling pointer checks?

Reply:
1) I would expect a fairly extensive comment as to why we are disabling pointer checks.
2) It seems @angelhof is rather calling for even stricter pointer checks? I believe we could use `__CPROVER_POINTER_OBJECT(foo) == __CPROVER_POINTER_OBJECT(bar)` to do the suggested comparison.

Reply:
We can't guarantee `__CPROVER_POINTER_OBJECT(buf->buffer) == __CPROVER_POINTER_OBJECT(ring_buffer->allocation)` because external code can call this function via `aws_ring_buffer_buf_belongs_to_pool`. We could make this a requirement at the API level, but then I think this function will always return true.

It is possible we could rewrite this function using pointer-to-integer casts. Although the guarantees for integers converted from pointers are implementation-defined.

Or we can scratch this up as being okay in practice and disable the pointer-check.

Reply:
So maybe do `if(__CPROVER_POINTER_OBJECT(buf->buffer) == __CPROVER_POINTER_OBJECT(ring_buffer->allocation)) { ...` instead of disabling the pointer checks completely?

Reply:
Ok, now we short-circuit if the pointer comparison is undefined due to different pointer objects.